pageserver/src/virtual_file.rs

```diff
@@ -566,12 +559,19 @@ impl VirtualFile {
 impl Drop for VirtualFile {
     /// If a VirtualFile is dropped, close the underlying file if it was open.
     fn drop(&mut self) {
-        let handle = self.handle.get_mut().unwrap();
-
-        // We could check with a read-lock first, to avoid waiting on an
-        // unrelated I/O.
+        let handle = self.handle.get_mut();
+
+        // We don't have async drop so we cannot wait for the lock here.
+        // Instead, do a best-effort attempt at closing the underlying
+        // file descriptor by using `try_write`.
```

Comment:
```suggestion
```

I think it's best to leave out speculation for likely not fitting features of the language.

Reply:
WDYM by not fitting? Why would async drop not fit here?

Reply:
My personal opinion of how it probably is not a good fit for rust language :)

Reply:
Let's just resolve this, fine to leave this.